### PR TITLE
Updated mutate_group to sort (arrange) result based on group columns

### PR DESCRIPTION
**Pull request by hidekoji**
# Description

Updated mutate_group to sort (arrange) result based on group columns

# Checklist
Make sure you have performed the following items before submitting this pull request.
If not, please describe the reason.  

- [ ] Add test cases for this fix/enhancement
- [ ] Pass devtools::check()
- [ ] Pass devtools::test()
- [ ] Test installing from github
- [x] Tested with Exploratory

### CODE REVIEW

File: R/util.R [hunk 1]
```diff
@@ -2329,12 +2329,12 @@ mutate_group <- function(.data, keep_group = FALSE, group_cols = NULL, group_fun
         name_list <- group_cols
       }
       names(groupby_args) <- name_list
-      # make sure to ungroup result
-      .data %>% dplyr::group_by(!!!groupby_args) %>% dplyr::mutate(...)
+      # make sure to sorr result by group by columns
```
Comment (hideaki) [inline R/util.R:2332]: Typo in the comment.

File: tests/testthat/test_util.R [hunk 2]
```diff
@@ -1230,9 +1230,9 @@ test_that("is_japanese_holiday", {
 
 test_that("mutate_group", {
   df <- mtcars %>% exploratory::mutate_group(group_cols = c(cyl="cyl", mpg_int10="mpg"), group_funs = c("none", "asintby10"), mpg_cummean = cummean(mpg))
-  expect_equal(head(df)$mpg_cummean[[1]],21)
+  expect_equal(head(df)$mpg_cummean[[1]],22.8)
```
Comment (hideaki) [inline tests/testthat/test_util.R:1233]: We could check some more columns and some more rows.

File: tests/testthat/test_util.R [hunk 3]
```diff
@@ -1230,9 +1230,9 @@ test_that("is_japanese_holiday", {
...
   df2 <- mtcars %>% exploratory::mutate_group(group_cols = c(cyl="cyl", mpg_int10="mpg"), group_funs = c("none", "asintby10"), wt_cummean = cummean(wt))
-  expect_equal(head(df2)$wt_cummean[[1]],2.62)
+  expect_equal(head(df2)$wt_cummean[[1]],2.32)
```
Comment (hideaki) [inline tests/testthat/test_util.R:1235]: We could check some more columns and some more rows.

Comment (hidekoji) [reply tests/testthat/test_util.R:1238]: Added more test case.

File: tests/testthat/test_util.R (hunk 2 above)
Comment (hidekoji) [reply tests/testthat/test_util.R:1233]: Added more test case.

File: R/util.R (hunk 1 above)
Comment (hidekoji) [reply R/util.R]: fixed typo.

File: R/util.R [hunk 4]
```diff
@@ -2329,12 +2329,12 @@ mutate_group <- function(.data, keep_group = FALSE, group_cols = NULL, group_fun
...
       names(groupby_args) <- name_list
-      # make sure to ungroup result
-      .data %>% dplyr::group_by(!!!groupby_args) %>% dplyr::mutate(...)
+      # make sure to sort result by group by columns
+      .data %>% dplyr::group_by(!!!groupby_args) %>% dplyr::mutate(...) %>% dplyr::arrange(!!!groupby_args)
     } else {
       if(!is.null(group_cols)) { # In case only group_by columns are provied, group_by with the columns
-        # make sure to ungroup result
-        .data %>% dplyr::group_by(!!!rlang::syms(group_cols)) %>% dplyr::mutate(...)
+        # make sure to sorr result by group by columns
```
Comment (hideaki) [inline R/util.R:2336]: Typo in the comment.

**Pull request merged**